jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/MavenSpyLogProcessor.java

```diff
@@ -201,6 +202,39 @@ public String toString() {
                     (file == null ? "" : " " + file) +
                     '}';
         }
+        
+        @Override
+        public int hashCode() {
+        	return Objects.hash(groupId, artifactId, baseVersion);
+        }
+        
+        @Override
+		public boolean equals(Object obj) {
+			if (this == obj)
+				return true;
+			if (obj == null)
+				return false;
+			if (getClass() != obj.getClass())
+				return false;
+			MavenArtifact other = (MavenArtifact) obj;
+			if (groupId == null) {
+				if (other.groupId != null)
+					return false;
+			} else if (!groupId.equals(other.groupId))
+				return false;
+			if (artifactId == null) {
+				if (other.artifactId != null)
+					return false;
+			} else if (!artifactId.equals(other.artifactId))
+				return false;
+			if (baseVersion == null) {
+				if (other.baseVersion != null)
+					return false;
+			} else if (!baseVersion.equals(other.baseVersion))
+				return false;
+			return true;
+		}
+        
     }
 
     public static class MavenDependency extends MavenArtifact {
```

Comment:
We will have to extract the domain objects soon :-)

I wanted to move to JAXB for the XML model and to be schema first (rather than code first) between the maven event spy and the jenkins plugin. 

I'm not sure how we could combine an Xsd2Java approach with these domain objects that would be more than POJOs and have domain methods. cc @alvarolobato